jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfig.java

```diff
@@ -76,10 +76,10 @@
    *     none exists
    */
   @Nullable
-  String getAuthFor(String registry) {
+  AuthTemplate getAuthFor(String registry) {
```

Comment:
should this be optional?

Reply:
Then I'd take out all `@Nullable`s from other methods in this class for consistency, but it becomes a bit involves. Besides, I think it's not that critical to migrate to Optional for this method, as it is not public.